nexus/src/external_api/http_entrypoints.rs

```diff
@@ -5083,6 +5084,97 @@ async fn updates_refresh(
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
 
+// Metrics
+
+#[derive(Debug, Deserialize, JsonSchema)]
+pub struct SystemMetricParams {
+    #[serde(flatten)]
+    pub pagination: dropshot::PaginationParams<
+        params::ResourceMetrics,
+        params::ResourceMetrics,
+    >,
+
+    /// The UUID of the container being queried
+    // TODO: I might want to force the caller to specify type here?
+    pub id: Uuid,
+}
+
+#[derive(Display, Deserialize, JsonSchema)]
+#[display(style = "snake_case")]
+#[serde(rename_all = "snake_case")]
+pub enum SystemMetricName {
+    VirtualDiskSpaceProvisioned,
+    CpusProvisioned,
+    RamProvisioned,
+}
+
+#[derive(Deserialize, JsonSchema)]
+struct SystemMetricsPathParam {
+    metric_name: SystemMetricName,
+}
+
+/// Access metrics data
+#[endpoint {
+     method = GET,
+     path = "/system/metrics/{metric_name}",
+     tags = ["system"],
+}]
+async fn system_metric(
+    rqctx: Arc<RequestContext<Arc<ServerContext>>>,
+    path_params: Path<SystemMetricsPathParam>,
+    query_params: Query<SystemMetricParams>,
+) -> Result<HttpResponseOk<ResultsPage<oximeter_db::Measurement>>, HttpError> {
+    let apictx = rqctx.context();
+    let nexus = &apictx.nexus;
+    let metric_name = path_params.into_inner().metric_name;
+
+    let query = query_params.into_inner();
+    let limit = rqctx.page_limit(&query.pagination)?;
+
+    let handler = async {
+        let opctx = OpContext::for_external_api(&rqctx).await?;
+
+        let result = match metric_name {
+            SystemMetricName::VirtualDiskSpaceProvisioned => {
+                opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
+                nexus
+                    .select_timeseries(
+                        "collection_target:virtual_disk_space_provisioned",
+                        &[&format!("id=={}", query.id)],
+                        query.pagination,
+                        limit,
+                    )
+                    .await?
+            }
+            SystemMetricName::CpusProvisioned => {
+                opctx.authorize(authz::Action::Read, &authz::FLEET).await?;
```

Comment:
Should these authz checks not be in the app layer?

Reply:
Done in e954c771e0572c3a8cb2ce68657b10211a68b7b1